### PR TITLE
feat: add development workflow for Accelerate

### PR DESCRIPTION
**Pull request by ankur-arch**
This section describes how it would look like to use Accelerate in development

### CODE REVIEW

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+title: 'Development workflow'
+metaTitle: 'Accelerate: Development environment'
```
Comment (nikolasburk) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: this page is very specifically about "local development", why not make that the title?

```suggestion
title: 'Local development'
metaTitle: 'Accelerate: Local development'
```

Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: The original title felt a bit too long. Your suggestion is much better 🙇 , thank you!

**Review by nikolasburk — APPROVED** [review]
I think this will be a very useful resource 🙌

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+metaDescription: 'Learn about Prisma Accelerate in a development environment.'
+tocDepth: 3
+toc: true
+---
+
+<TopBlock>
+
+Prisma Accelerate is designed to efficiently scale database connections in a production environment while providing a global cache. To leverage the benefits of connection pooling and a global cache, it is essential to utilize a publicly accessible database.
+
+In development environments, using a publicly available database may not be possible, and you may want to use a local database. Here, we will explain how to use Prisma Accelerate client extension in a development environment with a local database.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: > In development environments, using a publicly available database may not be possible,

Why is that?

Besides that, yes of course you want to use a local database for local testing. But for all kinds of other reasons.

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+</TopBlock>
+
+## Prisma Accelerate with a local database
+
+To benefit from Accelerate's connection pool and global cache features, you must have a publicly accessible database, enabling the creation of a Prisma Accelerate URL. It's important to note that Accelerate's features won't function with a local database.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: The first sentence here is a full repetition to what was already said above, and is not relevant to this headline at all. 
It is also not correct that Accelerate's features won't function with a local database. Accelerate does not work with a local database at all.

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+<img src="images/accelerate-in-dev.png" alt="" />
+
+However, using a local database URL, you can add the Accelerate client extension to your Prisma Client. This will allow you to use a local database in your development environment. Keep in mind that, in this scenario, caching and connection pooling won't be available.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: What does adding the client extension then do? Why not skip it completely?

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+<img src="images/accelerate-in-prod.png" alt="" />
+
+In a production environment, when a generated Accelerate connection string is used, you will be able to use connection pooling and caching by adding the Accelerate client extension to the `PrismaClient`.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: How is this sentence relevant to the headline?

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+3. Add Prisma Accelerate to your project
+
+   ```typescript
+   import { PrismaClient } from '@prisma/client'
+   import { withAccelerate } from '@prisma/extension-accelerate'
+
+   const prisma = new PrismaClient().$extends(withAccelerate())
+   ```
+
+   > `PrismaClient` infers that a local database is being used from the database connection string and disables the connection pooling and caching functionalities of Prisma Accelerate.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: As written before, this is not correct. It does not disable anything. It just does not use Prisma Accelerate at all as it uses the local database instead of Prisma Accelerate.

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+## Using Prisma Accelerate in edge functions in a development environment
+
+When working with Prisma Accelerate in a dev environment for [Next.js edge functions](https://vercel.com/docs/functions/edge-functions), you must use a `PrismaClient` without an engine.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: Why do you have to use Prisma without an engine in this situation?
What happens if you don't?

Comment (janpio) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: Also, `PrismaClient` is an implementation detail that is not relevant here. "Prisma" or "Prisma ORM" would be the correct wording here.

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+To create this engineless PrismaClient, run the command `prisma generate --no-engine`. Now, you will be able to use `PrismaClient` for the edge.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: You can also use the edge import without `--no-engine`.

Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: You always have to use the edge import for Accelerate so I am not clear why this should be a change to call out to someone using Accelerate already.

Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: "for the edge" is weird description, usually we say "in edge functions" as in the headline.

Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: I improved messaging in my latest [commit](https://github.com/prisma/docs/pull/5505/commits/0442259aa57010b38b7b4fec692a95835da3f36f)

Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: I suggested the best practice, I understand how my wording was misleading. I changed the content in the latest commit.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 5 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: I changed the headline to `Using Prisma Accelerate client extension in development and production`, that would make it more relevant to the content following the headline.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 4 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: I explained in more detail in why you wouldn't want to skip adding the client extension

```markdown
Prisma Accelerate is designed to efficiently scale database connections in a production environment while providing a global cache. To leverage the benefits of connection pooling and a global cache, it is essential to utilize a publicly accessible database.

In development environments, you may want to use a local database to minimize expenses.

Furthermore, you may consider extending `PrismaClient` with the Accelerate client extension once, so that you can use a local database in development and a hosted database with Accelerate’s connection pooling and caching enabled. This eliminates the need for conditional logic to switch clients between development and production.

Here, we will explain how to use Prisma Accelerate client extension in a development environment with a local database.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 2 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: Simplified the line to `In development environments, you may want to use a local database to minimize expenses.
`. Thanks for the callout 😄 .

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 3 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: I cleared out the repetition.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 6 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: Changed this section.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 7 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: I changed the content for this section aswell.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 8 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: Plus some callouts are explicit for these user [queries](https://prisma.slack.com/archives/C04KTN0V2Q0/p1701038300668449?thread_ts=1701036632.570479&cid=C04KTN0V2Q0).

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 9]
```diff
@@ -0,0 +1,65 @@
+---
+title: 'Local development'
+metaTitle: 'Accelerate: Local development'
+metaDescription: 'Learn about Prisma Accelerate in a development environment.'
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
metaDescription: 'Learn how to use Prisma Accelerate in a development environment.'
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+tocDepth: 3
+toc: true
+---
+
+<TopBlock>
+
+Prisma Accelerate is designed to efficiently scale database connections in a production environment while providing a global cache. To leverage the benefits of connection pooling and a global cache, it is essential to utilize a publicly accessible database.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
Prisma Accelerate is designed to scale database connections efficiently in production environments. Prisma Accelerate also provides a global cache to reduce the load on your database and reduce query response time. To leverage Prisma Accelerate, it is essential to utilize a publicly accessible database.
```

Feel free to adjust the text to what makes sence

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+In development environments, you may want to use a local database to minimize expenses. Furthermore, you may consider extending `PrismaClient` with the Accelerate client extension once, so that you can use a local database in development and a hosted database with Accelerate’s connection pooling and caching enabled. This eliminates the need for conditional logic to switch clients between development and production.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: I don't think we need this paragraph. (Up to your judgement)

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+Here, we will explain how to use Prisma Accelerate client extension in a development environment with a local database.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
This guide will explain how to use Prisma Accelerate client extension in a development environment with a local database.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+   DATABASE_URL="postgres://username:password@127.0.0.1:5432/localdb"
+   ```
+
+2. Generate a Prisma Client
+
+   ```bash
+   npx prisma generate
+   ```
+
+   > Note: The `--no-engine` cannot be used when working with a local database, as it generates an engineless Prisma Client which requires the use of an Accelerate connection string.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
   > Note: The `--no-engine` cannot be used when working with a local database. The command generates Prisma Client artifacts without a [Query Engine](/concepts/components/prisma-engines) file, which requires the use of an Accelerate connection string.
```

Feel free to adjust the text as you prefer.

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+3. Add Prisma Accelerate to your project
+
+   ```typescript
+   import { PrismaClient } from '@prisma/client'
+   import { withAccelerate } from '@prisma/extension-accelerate'
+
+   const prisma = new PrismaClient().$extends(withAccelerate())
+   ```
+
+   > `PrismaClient` with the Accelerate client extension uses the local database connection. Accelerate won’t work as a local database is being used.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
   > The extended instance of Prisma Client will use the local database. Prisma Accelerate will not be used in your development environment to respond to respond to your Prisma Client queries.
```

Feel free to adjust the text.

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+
+<img src="images/accelerate-in-prod.png" alt="" />
+
+In a production environment, when a generated Accelerate connection string is used, you will be able to use connection pooling and caching by adding the Accelerate client extension to the `PrismaClient`.
+
+## Locally using Prisma Accelerate in edge functions
+
+When working with Prisma Accelerate in a dev environment for [Next.js edge functions](https://vercel.com/docs/functions/edge-functions), you must use `PrismaClient` that can be used in edge functions.
+
+```bash
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
````suggestion
```typescript
````

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+
+   > `PrismaClient` with the Accelerate client extension uses the local database connection. Accelerate won’t work as a local database is being used.
+
+<img src="images/accelerate-in-prod.png" alt="" />
+
+In a production environment, when a generated Accelerate connection string is used, you will be able to use connection pooling and caching by adding the Accelerate client extension to the `PrismaClient`.
+
+## Locally using Prisma Accelerate in edge functions
+
+When working with Prisma Accelerate in a dev environment for [Next.js edge functions](https://vercel.com/docs/functions/edge-functions), you must use `PrismaClient` that can be used in edge functions.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
When using an [edge runtime](TODO), e.g., [Vercel edge functions](https://vercel.com/docs/functions/edge-functions), for your development environment, update your Prisma Client import as follows:
```

adjust the text as you see fit

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+
+```bash
+import { PrismaClient } from "@prisma/client/edge";
+```
+
+The `PrismaClient` in Next.js edge functions does not support a direct database connection, hence you have to use the Accelerate connection string.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
Edge function environments do not natively support TCP-based database connections. Prisma Accelerate provides a connection string that allows you to query your database over HTTP.
```

Not completely correct, adjust as you see fit.

**Review by ruheni — CHANGES_REQUESTED** [review]
Left a few suggestions. It mostly looks good to me, but not quite ready to be merged.

Quick note Next.js edge functions has been used interchangeably with Vercel edge functions. This is not technically correct.

**Review by ruheni — CHANGES_REQUESTED** [review]
Left a few suggestions. It mostly looks good to me, but not quite ready to be merged.

Quick note Next.js edge functions has been used interchangeably with Vercel edge functions. This is not technically correct.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 9 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: @ruheni, you can't use Accelerate in a development environment hence the meta description is like that.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 10 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: I like the suggested text.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 9 above)
Comment (ruheni) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
metaDescription: 'Learn how to set up Prisma Accelerate in a development environment.'
```
I think the metaDescription could be a little more specific. If the suggestions don't make sense, feel free to ignore the suggestions.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 14 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
   > The extended instance of Prisma Client will use the local database. Hence, Prisma Accelerate will not be used in your development environment to respond to your Prisma Client queries.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 11 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: I believe keeping this explains the problem better to our users. For example a user with a query similar to this [conversation](https://prisma.slack.com/archives/C04KTN0V2Q0/p1701366877198829?thread_ts=1696558562.734659&cid=C04KTN0V2Q0) would know the context of the page by reading the headline. cc: @ruheni  @jharrell thoughts?

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 17 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: Both Cloudflare and Deno Edge have their own APIs to connect directly to a database from an edge runtime.
I think the statement below should be a bit better without implying that TCP connections cannot be supported across all edge runtimes. 


```suggestion
While edge function environments generally lack native support for existing APIs enabling TCP-based database connections, Prisma Accelerate provides a connection string that allows querying your database over HTTP, a protocol supported in all edge runtimes.
```

cc: @ruheni

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 16 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
When using an edge runtime, e.g., [Next.js edge runtime](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes#edge-runtime), for your development environment, update your Prisma Client import as follows:
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+metaDescription: 'Learn how to use Prisma Accelerate in a development environment.'
+tocDepth: 3
+toc: true
+---
+
+<TopBlock>
+
+Prisma Accelerate is designed to scale database connections efficiently in production environments. Prisma Accelerate also provides a global cache to reduce the load on your database and reduce query response time. To leverage Prisma Accelerate, it is essential to utilize a publicly accessible database.
+
+In development environments, you may want to use a local database to minimize expenses. Furthermore, you may consider extending `PrismaClient` with the Accelerate client extension once, so that you can use a local database in development and a hosted database with Accelerate’s connection pooling and caching enabled. This eliminates the need for conditional logic to switch clients between development and production.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
In development environments, you may want to use a local database to minimize expenses. Furthermore, you may consider extending Prisma Client with the Accelerate client extension once so that you can use a local database in development and a hosted database with Accelerate’s connection pooling and caching enabled. This eliminates the need for conditional logic to switch clients between development and production.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+
+## Locally using Prisma Accelerate in edge functions
+
+When using an edge runtime, e.g., [Next.js edge runtime](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes#edge-runtime), for your development environment, update your Prisma Client import as follows:
+
+```typescript
+import { PrismaClient } from "@prisma/client/edge";
+```
+
+While edge function environments generally lack native support for existing APIs enabling TCP-based database connections, Prisma Accelerate provides a connection string that allows querying your database over HTTP, a protocol supported in all edge runtimes.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
Generally, edge function environments lack native support for existing APIs enabling TCP-based database connections.Prisma Accelerate provides a connection string that allows querying your database over HTTP, a protocol supported in all edge runtimes.
```

The smaller your sentences, the better. This would make it easier to read.

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+   import { withAccelerate } from '@prisma/extension-accelerate'
+
+   const prisma = new PrismaClient().$extends(withAccelerate())
+   ```
+
+   > The extended instance of Prisma Client will use the local database. Hence, Prisma Accelerate will not be used in your development environment to respond to your Prisma Client queries.
+
+<img src="images/accelerate-in-prod.png" alt="" />
+
+In a production environment, when a generated Accelerate connection string is used, you will be able to use connection pooling and caching by adding the Accelerate client extension to the `PrismaClient`.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
If an Accelerate connection string is used as the `DATABASE_URL` environment variable, Prisma Client will route your queries through Accelerate. We recommend using an Accelerate connection string in a production environment.
```

Feel free to adjust the text as you see fit

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+   DATABASE_URL="postgres://username:password@127.0.0.1:5432/localdb"
+   ```
+
+2. Generate a Prisma Client
+
+   ```bash
+   npx prisma generate
+   ```
+
+   > Note: The `--no-engine` cannot be used when working with a local database. The command generates Prisma Client artifacts without a [Query Engine](/concepts/components/prisma-engines) file, which requires the use of an Accelerate connection string.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
   > Note: The `--no-engine` flag should only be used in preview and production environments. The command generates Prisma Client artifacts without a [Query Engine](/concepts/components/prisma-engines) file, which requires an Accelerate connection string.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 20 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]: @ruheni if they are using a Next.js edge runtime, they have no option but to use Accelerate as of now.

Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
If an Accelerate connection string is used as the `DATABASE_URL` environment variable, Prisma Client will route your queries through Accelerate.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+This guide will explain how to use Prisma Accelerate client extension in a development environment with a local database.
+
+</TopBlock>
+
+## Using Prisma Accelerate client extension in development and production
+
+<br />
+<img src="images/accelerate-in-dev.png" alt="" />
+
+Accelerate doesn’t work with a local database. However, using a local database URL, you can add the Accelerate client extension to your Prisma Client. This will allow you to use a local database in your development environment. Keep in mind that, in this scenario, caching and connection pooling won't be available.
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
Accelerate does not work with a local database. However, in a development environment, you can still use Prisma Client with the Accelerate client extension. This setup will not include Accelerate's connection pooling and caching features.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+
+Here is how to use a local PostgreSQL database with Prisma ORM and Prisma Accelerate client extension (without caching and connection pooling):
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
Here is how to use a local PostgreSQL database with Prisma ORM and Prisma Accelerate client extension:
```

Since that was mentioned in the previous paragraph, we can remove it

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+
+1. Paste your local database URL in your .env file
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
1. Update the `DATABASE_URL` environment variable with your local database's connection string:
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+
+   ```.env
+   DATABASE_URL="postgres://username:password@127.0.0.1:5432/localdb"
+   ```
+
+2. Generate a Prisma Client
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
2. Generate a Prisma Client:
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+
+   ```bash
+   npx prisma generate
+   ```
+
+   > Note: The `--no-engine` flag should only be used in preview and production environments. The command generates Prisma Client artifacts without a [Query Engine](/concepts/components/prisma-engines) file, which requires an Accelerate connection string.
+
+3. Add Prisma Accelerate to your project
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
3. Set up Prisma Accelerate to your project:
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+</TopBlock>
+
+## Using Prisma Accelerate client extension in development and production
+
+<br />
+<img src="images/accelerate-in-dev.png" alt="" />
+
+Accelerate does not work with a local database. However, in a development environment, you can still use Prisma Client with the Accelerate client extension. This setup will not include Accelerate's connection pooling and caching features.
+
+Here is how to use a local PostgreSQL database with Prisma ORM and Prisma Accelerate client extension:
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
The following steps outline how to use Prisma ORM and Prisma Accelerate with a local PostgreSQL database.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+   const prisma = new PrismaClient().$extends(withAccelerate())
+   ```
+
+   > The extended instance of Prisma Client will use the local database. Hence, Prisma Accelerate will not be used in your development environment to respond to your Prisma Client queries.
+
+<img src="images/accelerate-in-prod.png" alt="" />
+
+If an Accelerate connection string is used as the `DATABASE_URL` environment variable, Prisma Client will route your queries through Accelerate.
+
+## Locally using Prisma Accelerate in edge functions
```
Comment (ruheni) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
## Using Prisma Accelerate locally using an edge functions runtime
```

Feel free to adjust the text

Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
##  Using Prisma Accelerate locally using an edge runtime
```

edge functions runtime feels a bit weird to me.

**Review by ruheni — APPROVED** [review]

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+This guide will explain how to use Prisma Accelerate client extension in a development environment with a local database.
+
+</TopBlock>
+
+## Using Prisma Accelerate client extension in development and production
+
+<br />
+<img src="images/accelerate-in-dev.png" alt="" />
+
+Accelerate does not work with a local database. However, in a development environment, you can still use Prisma Client with the Accelerate client extension. This setup will not include Accelerate's connection pooling and caching features.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
Accelerate does not work with a local database. However, in a development environment, you can still use Prisma Client with the Accelerate client extension. This setup will not provide Accelerate's connection pooling and caching features.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 30]
```diff
@@ -0,0 +1,65 @@
...
+
+2. Generate a Prisma Client:
+
+   ```bash
+   npx prisma generate
+   ```
+
+   > Note: The `--no-engine` flag should only be used in preview and production environments. The command generates Prisma Client artifacts without a [Query Engine](/concepts/components/prisma-engines) file, which requires an Accelerate connection string.
+
+3. Set up Prisma Accelerate to your project:
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: This sounds weird grammatically.

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 31]
```diff
@@ -0,0 +1,65 @@
...
+
+   ```typescript
+   import { PrismaClient } from '@prisma/client'
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx:45]: In usual Accelerate deployments this require `@prisma/client/edge`, right? 
So you still need different code for local development?

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 32]
```diff
@@ -0,0 +1,65 @@
...
+
+##  Using Prisma Accelerate locally using an edge runtime
+
+When using an edge runtime, e.g., [Next.js edge runtime](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes#edge-runtime), for your development environment, update your Prisma Client import as follows:
+
+```typescript
+import { PrismaClient } from "@prisma/client/edge";
+```
+
+Generally, edge function environments lack native support for existing APIs enabling TCP-based database connections.Prisma Accelerate provides a connection string that allows querying your database over HTTP, a protocol supported in all edge runtimes.
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
Generally, edge function environments lack native support for existing APIs enabling TCP-based database connections. Prisma Accelerate provides a connection string that allows querying your database over HTTP, a protocol supported in all edge runtimes.
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx [hunk 33]
```diff
@@ -0,0 +1,65 @@
...
+
+   > The extended instance of Prisma Client will use the local database. Hence, Prisma Accelerate will not be used in your development environment to respond to your Prisma Client queries.
+
+<img src="images/accelerate-in-prod.png" alt="" />
+
+If an Accelerate connection string is used as the `DATABASE_URL` environment variable, Prisma Client will route your queries through Accelerate.
+
+##  Using Prisma Accelerate locally using an edge runtime
+
+When using an edge runtime, e.g., [Next.js edge runtime](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes#edge-runtime), for your development environment, update your Prisma Client import as follows:
```
Comment (janpio) [inline content/800-data-platform/100-accelerate/580-local-development.mdx]: "edge runtime" is usually not how we call this, but "edge functions". The runtime is an underlying implementation detail.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 31 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx:45]: > In usual Accelerate deployments this require @prisma/client/edge, right?
So you still need different code for local development?

You'd need the @prisma/client/edge for edge function deployments, which has a dedicated section below 😄 . This is the case where they don't have to use the edge client.

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 33 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
##  Using Prisma Accelerate locally in an edge function

When using an edge function, e.g., [Vercel's edge functions](https://vercel.com/docs/functions/edge-functions), for your development environment, update your Prisma Client import as follows:
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 30 above)
Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx]:
```suggestion
3. Set up Prisma Client with the Accelerate client extension:
```

File: content/800-data-platform/100-accelerate/580-local-development.mdx (hunk 31 above)
Comment (janpio) [reply content/800-data-platform/100-accelerate/580-local-development.mdx:45]: Isn't the "no changes between production and local development needed" statement then wrong?

Comment (ankur-arch) [reply content/800-data-platform/100-accelerate/580-local-development.mdx:45]: @janpio  Do you mean the following statement at the beginning of the text?

> This eliminates the need for conditional logic to switch clients between development and production.

When using a regular Node.js runtime, this statement is true.

In the case of an edge function, as of now, you'd have to use the Prisma edge client, which requires the use of Accelerate. Hence the original statement does not apply to this edge case. 

Do you feel that the copy edit below sounds better/more specific

> This eliminates the need for conditional logic to switch clients between development and production (except for edge functions).
